azkaban-web-server/src/main/java/azkaban/imagemgmt/daos/ImageTypeDaoImpl.java

```diff
@@ -0,0 +1,113 @@
+      "insert into image_ownerships ( type_id, owner, role, created_on, created_by ) "
+          + "values (?, ?, ?, ?, ?)";
+
+  @Inject
+  public ImageTypeDaoImpl(DatabaseOperator databaseOperator) {
+    this.databaseOperator = databaseOperator;
+  }
+
+  @Override
+  public int createImageType(ImageType imageType) {
+    final SQLTransaction<Integer> insertAndGetSpaceId = transOperator -> {
+      String currentTime = DateTime.now().toLocalDateTime().toString();
+      transOperator.update(INSERT_IMAGE_TYPE, imageType.getType(), imageType.getDescription(),
+          true, imageType.getDeployable().getName(), currentTime, imageType.getCreatedBy());
+      int imageTypeId = Long.valueOf(transOperator.getLastInsertId()).intValue();
+      if( imageType.getOwnerships() != null && imageType.getOwnerships().size() > 0) {
+        for(ImageOwnership imageOwnership : imageType.getOwnerships()) {
+          transOperator.update(INSERT_IMAGE_OWNERSHIP, imageTypeId, imageOwnership.getOwner(),
+              imageOwnership.getRole(), currentTime, imageType.getCreatedBy());
+        }
+      }
+      transOperator.getConnection().commit();
+      return imageTypeId;
+    };
+
+    int imageTypeId = 0;
+    try {
+      /* what will happen if there is a partial failure in
+         any of the below statements?
+         Ideally all should happen in a transaction */
+      imageTypeId = databaseOperator.transaction(insertAndGetSpaceId);
+      log.info("Created image type id :"+imageTypeId);
+    } catch (SQLException e) {
+      log.error("Unable to create the image type metadata", e);
+    }
+    return imageTypeId;
+  }
+
+  @Override
+  public Optional<ImageType> getImageTypeByType(String type) {
+    final FetchImageTypeHandler fetchImageTypeHandler = new FetchImageTypeHandler();
+    List<ImageType> imageTypes = new ArrayList<>();
+    try {
+      imageTypes = this.databaseOperator
+          .query(FetchImageTypeHandler.FETCH_IMAGE_TYPE_BY_TYPE, fetchImageTypeHandler, type);
+    } catch (final SQLException ex) {
+      log.error(FetchImageTypeHandler.FETCH_IMAGE_TYPE_BY_TYPE + " failed.", ex);
+      throw new RuntimeException("Unable to fetch image type metadata from image type : "+type);
+    }
+    return imageTypes.isEmpty() ? Optional.empty() : Optional.of(imageTypes.get(0));
+  }
+
+  public static class FetchImageTypeHandler implements ResultSetHandler<List<ImageType>> {
+    private static final String FETCH_IMAGE_TYPE_BY_ID =
+        "SELECT id, type, description, active, deployable, created_on, created_by FROM image_types"
+            + " WHERE id = ?";
+    private static final String FETCH_IMAGE_TYPE_BY_TYPE =
+        "SELECT id, type, description, active, deployable, created_on, created_by FROM image_types"
+            + " WHERE type = ?";
+    private static final String FETCH_ALL_IMAGE_TYPES =
+        "SELECT id, type, description, active, deployable, created_on, created_by FROM image_types";
```

Comment:
Can we please have fetch for active image types?

Reply:
Done